test/exclude-members.test.js

```diff
@@ -174,3 +175,133 @@ test('add and remove a person, post on the new feed', async (t) => {
+  // bob gets added and is removed
+  // carol stays in the group
+  const alice = Testbot({
+    keys: ssbKeys.generate(null, 'alice'),
+    mfSeed: Buffer.from(
+      '000000000000000000000000000000000000000000000000000000000000a1ce',
+      'hex'
+    ),
+  })
+  const bob = Testbot({
+    keys: ssbKeys.generate(null, 'bob'),
+    mfSeed: Buffer.from(
+      '0000000000000000000000000000000000000000000000000000000000000b0b',
+      'hex'
+    ),
+  })
+  const carol = Testbot({
+    keys: ssbKeys.generate(null, 'carol'),
+    mfSeed: Buffer.from(
+      '00000000000000000000000000000000000000000000000000000000000ca201',
+      'hex'
+    ),
+  })
+
+  await alice.tribes2.start()
+  await bob.tribes2.start()
+  await carol.tribes2.start()
+  t.pass('tribes2 started for everyone')
+
+  await p(alice.metafeeds.findOrCreate)()
+  const bobRoot = await p(bob.metafeeds.findOrCreate)()
+  const carolRoot = await p(carol.metafeeds.findOrCreate)()
+
+  await replicate(alice, bob)
+  await replicate(alice, carol)
+  await replicate(bob, carol)
+  t.pass('everyone replicates their trees')
+
+  const { id: groupId, writeKey: writeKey1 } = await alice.tribes2
+    .create()
+    .catch((err) => t.error(err, 'alice failed to create group'))
+
+  await replicate(alice, carol)
+
+  await alice.tribes2
+    .addMembers(groupId, [bobRoot.id, carolRoot.id])
+    .catch((err) => t.error(err, 'add bob fail'))
+
+  await replicate(alice, carol)
+
+  await carol.tribes2.acceptInvite(groupId)
+
+  await replicate(alice, carol)
+
+  const {
+    value: { author: firstFeedId },
+  } = await carol.tribes2
+    .publish({
+      type: 'test',
+      text: 'first post',
+      recps: [groupId],
+    })
+    .catch((err) => t.error(err, 'carol failed to publish on first feed'))
+  if (firstFeedId) t.pass('carol posted first post')
+
+  await replicate(alice, carol).catch(t.error)
+
+  const excludeMsg = await alice.tribes2
+    .excludeMembers(groupId, [bobRoot.id])
+    .then((res) => {
+      t.pass('alice excluded bob')
+      return res
+    })
+    .catch((err) => t.error(err, 'remove member fail'))
+
+  await replicate(alice, carol).catch(t.error)
+
+  // TODO: maybe remove?
+  await p(setTimeout)(10000)
+
+  const carolHasExcludeMsg = await p(carol.db.getMsg)(excludeMsg.key)
+
+  console.log('carol has exclude', carolHasExcludeMsg)
+
+  const {
+    value: { author: secondFeedId },
+  } = await carol.tribes2
+    .publish({
+      type: 'test',
+      text: 'second post',
+      recps: [groupId],
+    })
+    .catch(t.fail)
+  if (secondFeedId) t.pass('carol posted second post')
+
+  t.notEquals(
+    secondFeedId,
+    firstFeedId,
+    'feed for second publish is different to first publish'
+  )
+
+  const { writeKey: writeKey2 } = await carol.tribes2.get(groupId)
+
+  const branches = await pull(
+    carol.metafeeds.branchStream({ root: carolRoot, old: true, live: false }),
+    pull.collectAsPromise()
+  )
```

Comment:
after basing on #85 this

```js
  const branches = await pull(
    carol.metafeeds.branchStream({ root: carolRoot, old: true, live: false }),
    pull.collectAsPromise()
  )

  console.log('carol branches', branches)
```

logs 

```js
carol branches [
  [
    FeedDetails {
      id: [FeedDetails],
      parent: null,
      purpose: 'root',
      feedFormat: 'bendybutt-v1',
      seed: null,
      keys: null,
      recps: null,
      metadata: {},
      tombstoned: false,
      tombstoneReason: null
    }
  ]
]
```

even though carol has successfully posted twice to the group, including on different feeds

Reply:
realized that carolRoot is an object and not the root id. strange that it returns anything at all

Reply:
made related pr https://github.com/ssbc/ssb-meta-feeds/pull/122